Optimize ZERO_FILL flag to mkstring()

We can use the "*" printf specifier.

Signed-off-by: Petr Tesarik <ptesarik@suse.cz>

---
 tools.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/tools.c
+++ b/tools.c
@@ -1562,10 +1562,7 @@ mkstring(char *s, int size, ulong flags,
 	case LONG_HEX:
 		sprintf(s, "%lx", (ulong)opt);
 		break;
 	case (LONG_HEX|ZERO_FILL):
-		if (VADDR_PRLEN == 8)
-			sprintf(s, "%08lx", (ulong)opt);
-		else if (VADDR_PRLEN == 16)
-			sprintf(s, "%016lx", (ulong)opt);
+		sprintf(s, "%0*lx", VADDR_PRLEN, (ulong)opt);
 		break;
 	case INT_DEC:
 		sprintf(s, "%u", (uint)((ulong)opt));
